src/util/paths.py

```diff
@@ -14,11 +15,45 @@ def walk_up_search(filename: str) -> Optional[Path]:
     Returns:
         The path to the file, if found, or None.
     """
-    current = Path(os.getcwd())
+    current = Path.cwd()
     while True:
         if str(current) == current.root:
             return None
         filepath = current / filename
         if filepath.exists():
             return filepath
         current = current.parent
+
+
+def resolve_relative_path(start: Path, path: Path) -> Optional[Path]:
+    """Takes a starting point and attempts to resolve a relative path.
+
+    Args:
+        start: an absolute path to start from.
+        path: a relative path to resolve.
+
+    Returns:
+        An absolute path for the given relative path if valid, or None.
+    """
+    resolved = start / path
+
+    if resolved.exists() is None:
+        try:
+            resolved.mkdir(parents=True, exist_ok=True)
+        except PermissionError as error:
```

Comment:
You don't have to care about the specific failure mode. If it fails, it fails. Done. `raise Exception('i explodey uwu oh no')`